src/components/Footer.tsx

```diff
@@ -0,0 +1,100 @@
+import classNames from 'classnames';
+import React from 'react';
+import { Todo, SortType } from '../Types';
+import { client } from '../utils/client';
+
+interface Props {
+  numberOfActiveTodos: number,
+  selectedTab: SortType,
+  setSelectedTab: React.Dispatch<React.SetStateAction<SortType>>,
+  isThereCompletedTodos: boolean,
+  todo: Todo[],
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>,
+  setDeleteErrorMessage: React.Dispatch<React.SetStateAction<string>>,
+  setIsEveryThingDelete: React.Dispatch<React.SetStateAction<boolean>>,
+}
+
+export const Footer:React.FC<Props> = ({
+  numberOfActiveTodos,
+  selectedTab,
+  setSelectedTab,
+  isThereCompletedTodos,
+  todo,
+  setTodos,
+  setDeleteErrorMessage,
+  setIsEveryThingDelete,
+}) => {
+  const deleteCompletedTodos = async () => {
+    setIsEveryThingDelete(true);
+
+    const completedTodoIds = todo
+      .filter((element) => element.completed)
+      .map((element) => element.id);
+
+    try {
+      await Promise.all(
+        completedTodoIds.map((id) => client.delete(`/todos/${id}`)),
+      );
+
+      setTodos((prevTodo) => prevTodo.filter((element) => !element.completed));
+    } catch (error) {
+      setDeleteErrorMessage('Unable to delete completed todos');
+      throw Error('There is an issue deleting completed todos.');
+    } finally {
+      setIsEveryThingDelete(false);
```

Comment:
The setIsEveryThingDelete function call should be changed to setIsEverythingDeleted for better readability and consistency.